cdap-master/src/main/java/co/cask/cdap/data/tools/ArchiveUpgrader.java

```diff
@@ -0,0 +1,93 @@
+/*
+* Copyright © 2015 Cask Data, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not
+* use this file except in compliance with the License. You may obtain a copy of
+* the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations under
+* the License.
+*/
+
+package co.cask.cdap.data.tools;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.io.Locations;
+import co.cask.cdap.logging.LoggingConfiguration;
+import com.google.inject.Inject;
+import org.apache.twill.filesystem.Location;
+import org.apache.twill.filesystem.LocationFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+
+/**
+ * Class to Upgrade the archive directories
+ */
+public class ArchiveUpgrader extends AbstractUpgrader {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ArchiveUpgrader.class);
+  private final CConfiguration cConf;
+
+  @Inject
+  private ArchiveUpgrader(LocationFactory locationFactory, CConfiguration cConf) {
+    super(locationFactory);
+    this.cConf = cConf;
+  }
+
+  @Override
+  public void upgrade() throws Exception {
+    upgradeStream();
+    upgradeLogs();
+  }
+
+  /**
+   * Upgrades the streams archive path
+   */
+  private void upgradeStream() throws IOException {
+    LOG.info("Upgrading stream files ...");
```

Comment:
Yes its a part of MDSUpgrade as the data in app.meta table. It will come in a independent PR